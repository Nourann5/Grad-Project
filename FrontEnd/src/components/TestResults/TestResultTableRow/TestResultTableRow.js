import React, { useState } from 'react'
import styles from './TestResultTableRow.module.css'
import {ReactComponent as DownArrowIcon} from 'assets/icons/downArrowResults.svg'
import { Collapse } from 'react-bootstrap';
function TestResultTableRow({user}) {
    const [open, setOpen] = useState(false);
  return (
    <>
        <tr key={user?.id+user?.user_name+user?.user_email}>
            <th scope="row">{user?.id}</th>
            <td>B</td>
            <td>A</td>
            <td>
                {
                    <>
                        <p className={`${styles['table__user-status']} ${user?.status ==0?'':'d-none'}`}>
                            <span className={`${styles['table__user-status-color']} ${styles['table__user-status-color--new']}`}></span>
                            <span className={styles['table__user-status-value']}>Not Answered</span>
                        </p>
                        <p className={`${styles['table__user-status']} ${user?.status ==1?'':'d-none'}`}>
                            <span className={`${styles['table__user-status-color']} ${styles['table__user-status-color--active']}`}></span>
                            <span className={styles['table__user-status-value']}>Correct</span>
                        </p>
                        <p className={`${styles['table__user-status']} ${user?.status ==2?'':'d-none'}`}>
                            <span className={`${styles['table__user-status-color']} ${styles['table__user-status-color--blocked']}`}></span>
                            <span className={styles['table__user-status-value']}>Wrong</span>
                        </p>
                    </>
                }
            </td>
            <td>
                <button
                    onClick={() => setOpen(!open)}
                    aria-controls="example-collapse-text"
                    aria-expanded={open}
                >
                    <DownArrowIcon className={styles['table__user-down-arrow']}/>
                </button>
            </td>
        </tr>
        <tr>
            <td colSpan={5} className={`p-0`}>
                <Collapse in={open}>
                    <div>
                        <div className={styles['table__row-collapsed']}>
                            <div className={`${styles['test-result__table-collapse-info']} ${user?.status ==1?styles['test-result__table-collapse-info--correct']:styles['test-result__table-collapse-info--not-correct']}`}>
                                <h2 className={styles['test__info-heading']}>
                                    Questions 1-10 are based on the following passage.
                                </h2>
                                <p>
                                    Akira came directly, breaking all tradition. Was that it? Had he followed form had he asked his mother to speak to his father to approach a Line go-between would Chie have been more receptive? He came on a winter’s eve. He pounded on the door while a cold rain beat on the shuttered veranda, so at first Chie thought him only the wind. The maid knew better. Chie heard her soft scuttling footsteps, the creak of the door. Then the maid brought a 10 calling card to the drawing room, for Chie.
                                </p>
                                <p>
                                    Chie was reluctant to go to her guest; perhaps she was feeling too cozy. She and Naomi were reading at a low table set atop a charcoal brazier. A thick quilt spread over the sides of the table so their legs were 15 tucked inside with the heat. “Who is it at this hour, in this weather?” Chie questioned as she picked the name card off the maid’s lacquer tray.
                                </p>
                                <p>
                                    Chie was reluctant to go to her guest; perhaps she was feeling too cozy. She and Naomi were reading at a low table set atop a charcoal brazier. A thick quilt spread over the sides of the table so their legs were 15 tucked inside with the heat. “Who is it at this hour, in this weather?” Chie questioned as she picked the name card off the maid’s lacquer tray.
                                </p>
                                <p>
                                    Chie was reluctant to go to her guest; perhaps she was feeling too cozy. She and Naomi were reading at a low table set atop a charcoal brazier. A thick quilt spread over the sides of the table so their legs were 15 tucked inside with the heat. “Who is it at this hour, in this weather?” Chie questioned as she picked the name card off the maid’s lacquer tray.
                                </p>
                            </div>
                            <div className={`${styles['test-result__table-collapse-question']} ${user?.status ==1?styles['test-result__table-collapse-question--correct']:styles['test-result__table-collapse-question--not-correct']}`}>
                                <h2 className={styles['test__question-heading']}>
                                    Which choice best describes what happens in the passage?
                                </h2>
                    
                                <div className="test__input-answer">
                                {/* <input type="text" className="form-control" name="test__input-answer1" placeholder=""ظ> */}
                                </div>
                    
                                <div className={styles["test__choices-wrapper"]}>
                                    <div className={`${styles['test__choice-wrapper']} ${styles['test__choice-wrapper--correct']}`}>
                                        <span className={styles["test__choice-alpha"]}>A- </span>One Character Argues With Another Character Who Intrudes On  
                                    </div>
                                    <div className={`${styles['test__choice-wrapper']}`}>
                                        <span className={styles["test__choice-alpha"]}>B- </span>One Character Argues With Another Character Who Intrudes On  
                                    </div>
                                    <div className={`${styles['test__choice-wrapper']} ${styles['test__choice-wrapper--not-correct']}`}>
                                        <span className={styles["test__choice-alpha"]}>C- </span>One Character Argues With Another Character Who Intrudes On  
                                    </div>
                                    <div className={`${styles['test__choice-wrapper']}`}>
                                        <span className={styles["test__choice-alpha"]}>D- </span>One Character Argues With Another Character Who Intrudes On  
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>

                </Collapse>
            </td>
        </tr>
    </>
  )
}

export default TestResultTableRow