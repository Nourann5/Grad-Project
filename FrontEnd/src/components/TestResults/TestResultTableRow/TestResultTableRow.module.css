
.table thead tr{
    border-bottom: 1px solid #e4e4e4!important;
}
.table thead th{
    color: white;
    /* font-size: 14px; */
    background-color:var(--main-color)
}
.table thead th:first-of-type{
    border-top-left-radius: 8px;
    border-bottom-left-radius: 8px;
}
.table thead th:last-of-type{
    border-top-right-radius: 8px;
    border-bottom-right-radius: 8px;
}
.table tbody tr{
    border-bottom: 1px solid #e4e4e4!important;
    background-color: white!important;
}
.table tbody td{
    vertical-align: middle;
}
.table tbody tr:nth-child(2n+1){
    background-color: white!important;
}
.table tbody th{
    vertical-align: middle;
}
.table__row-collapsed{
    padding: 0.5rem 0.5rem;
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-gap: 20px;
    transition: height .3s linear!important;
}
.test-result__table-collapse-info{
    max-height: 320px;
    min-height: 100px;
    overflow: auto;
    padding: 30px;
    background-color: #EF2A350D;
    border-radius: 0;
}
.test-result__table-collapse-question{
    max-height: 320px;
    min-height: 100px;
    overflow: auto;
    padding: 30px;
    border-radius: 0;
}
.test-result__table-collapse-info--not-correct{
    background-color: #EF2A350D;
}
.test-result__table-collapse-info--correct{
    background-color: #26DA6226;
}
.test-result__table-collapse-question--not-correct{
    background-color: #EF2A350D;
}
.test-result__table-collapse-question--correct{
    background-color: #26DA6226;
}
.test__info-heading,.test__question-heading{
    font-size: 18px;
    font-family: 'PoppinsSemiBold';
    margin-bottom: 20px;
}
.test__choice-wrapper{
    display: flex;
    margin-bottom: 15px;
}
.test__choice-alpha{
    margin-right: 10px;
}
.test__choice-wrapper--correct{
    color: #26DA62;
}
.test__choice-wrapper--not-correct{
    color: #EF2A35;
}
.table__user-img-wrapper{
    display: flex;
    align-items: center;
}
.table__user-img{
    width: 30px;
    height: 30px;
    border-radius: 50%;
    margin-right: 10px;
}
.table__user-name{
    font-size: 16px;
    margin: 0;
    line-height: 16px;
    margin-bottom: 2px;
}
.table__user-email{
    font-size: 14px;
    color: #737171;
    line-height: 16px;
}
.table__user-status-cont{

}
.table__user-status-buttons{
    margin-right: 5px;
    border-radius: 3px;
    color: white;
    padding: 7px 15px;
}
.table__user-status-buttons--active{
    background-color: #3AC47D;
}
.table__user-status-buttons--blocked{
    background-color: #E15173;
}
.table__user-status-buttons--not-active{
    background-color: #dcdcdc;
    color: black;
}

.table__user-status{
    display: flex;
    align-items: center;
    /* justify-content: center; */
}
.table__user-status-color{
    width:20px;
    height:20px;
    border-radius:50%;
    margin-right: 5px;
}
.table__user-status-color--active{
    background-color:#26DA62
}
.table__user-status-color--blocked{
    background-color:#EF2A35
}
.table__user-status-color--new{
    background-color:#C3C5CB

}
.table__user-status-value{
    font-size: 14px;
    font-family: 'PoppinsMedium';
}
.table__user-action-button{

}
.table__user-action-cont{
    display: flex;
    align-items: center;
}
.table__user-action-icon{
    width:20px;
    height:20px;
}
.table__user-action-icon--edit{
    color:var(--main-color)
}
.table__user-action-icon--delete path{
    fill:#E15173
}
.table__user-group{
    width: max-content;
}
@media (max-width: 426px){
    .table__row-collapsed{
        grid-template-columns:1fr 
    }
}