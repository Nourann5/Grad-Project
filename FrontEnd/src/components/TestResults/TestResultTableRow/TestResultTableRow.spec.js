import {render , screen} from '@testing-library/react'
import { BrowserRouter, Route } from 'react-router-dom'
import TestResultTableRow from './TestResultTableRow'

describe('Test the Test Reuslt Table', () => { 
    it('heading exists',()=>{
        const tree =  render(
            <BrowserRouter>
                <TestResultTableRow/>
            </BrowserRouter>
        )
        expect(tree).toMatchSnapshot()
    })
 })