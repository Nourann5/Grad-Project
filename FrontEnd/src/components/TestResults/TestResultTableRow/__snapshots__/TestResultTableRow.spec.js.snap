// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Test Reuslt Table heading exists 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr>
        <th
          scope="row"
        />
        <td>
          B
        </td>
        <td>
          A
        </td>
        <td>
          <p
            class="table__user-status d-none"
          >
            <span
              class="table__user-status-color table__user-status-color--new"
            />
            <span
              class="table__user-status-value"
            >
              Not Answered
            </span>
          </p>
          <p
            class="table__user-status d-none"
          >
            <span
              class="table__user-status-color table__user-status-color--active"
            />
            <span
              class="table__user-status-value"
            >
              Correct
            </span>
          </p>
          <p
            class="table__user-status d-none"
          >
            <span
              class="table__user-status-color table__user-status-color--blocked"
            />
            <span
              class="table__user-status-value"
            >
              Wrong
            </span>
          </p>
        </td>
        <td>
          <button
            aria-controls="example-collapse-text"
            aria-expanded="false"
          >
            <svg
              class="table__user-down-arrow"
            >
              downArrowResults.svg
            </svg>
          </button>
        </td>
      </tr>
      <tr>
        <td
          class="p-0"
          colspan="5"
        >
          <div
            class="collapse"
          >
            <div
              class="table__row-collapsed"
            >
              <div
                class="test-result__table-collapse-info test-result__table-collapse-info--not-correct"
              >
                <h2
                  class="test__info-heading"
                >
                  Questions 1-10 are based on the following passage.
                </h2>
                <p>
                  Akira came directly, breaking all tradition. Was that it? Had he followed form had he asked his mother to speak to his father to approach a Line go-between would Chie have been more receptive? He came on a winter’s eve. He pounded on the door while a cold rain beat on the shuttered veranda, so at first Chie thought him only the wind. The maid knew better. Chie heard her soft scuttling footsteps, the creak of the door. Then the maid brought a 10 calling card to the drawing room, for Chie.
                </p>
                <p>
                  Chie was reluctant to go to her guest; perhaps she was feeling too cozy. She and Naomi were reading at a low table set atop a charcoal brazier. A thick quilt spread over the sides of the table so their legs were 15 tucked inside with the heat. “Who is it at this hour, in this weather?” Chie questioned as she picked the name card off the maid’s lacquer tray.
                </p>
                <p>
                  Chie was reluctant to go to her guest; perhaps she was feeling too cozy. She and Naomi were reading at a low table set atop a charcoal brazier. A thick quilt spread over the sides of the table so their legs were 15 tucked inside with the heat. “Who is it at this hour, in this weather?” Chie questioned as she picked the name card off the maid’s lacquer tray.
                </p>
                <p>
                  Chie was reluctant to go to her guest; perhaps she was feeling too cozy. She and Naomi were reading at a low table set atop a charcoal brazier. A thick quilt spread over the sides of the table so their legs were 15 tucked inside with the heat. “Who is it at this hour, in this weather?” Chie questioned as she picked the name card off the maid’s lacquer tray.
                </p>
              </div>
              <div
                class="test-result__table-collapse-question test-result__table-collapse-question--not-correct"
              >
                <h2
                  class="test__question-heading"
                >
                  Which choice best describes what happens in the passage?
                </h2>
                <div
                  class="test__input-answer"
                />
                <div
                  class="test__choices-wrapper"
                >
                  <div
                    class="test__choice-wrapper test__choice-wrapper--correct"
                  >
                    <span
                      class="test__choice-alpha"
                    >
                      A- 
                    </span>
                    One Character Argues With Another Character Who Intrudes On
                  </div>
                  <div
                    class="test__choice-wrapper"
                  >
                    <span
                      class="test__choice-alpha"
                    >
                      B- 
                    </span>
                    One Character Argues With Another Character Who Intrudes On
                  </div>
                  <div
                    class="test__choice-wrapper test__choice-wrapper--not-correct"
                  >
                    <span
                      class="test__choice-alpha"
                    >
                      C- 
                    </span>
                    One Character Argues With Another Character Who Intrudes On
                  </div>
                  <div
                    class="test__choice-wrapper"
                  >
                    <span
                      class="test__choice-alpha"
                    >
                      D- 
                    </span>
                    One Character Argues With Another Character Who Intrudes On
                  </div>
                </div>
              </div>
            </div>
          </div>
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr>
      <th
        scope="row"
      />
      <td>
        B
      </td>
      <td>
        A
      </td>
      <td>
        <p
          class="table__user-status d-none"
        >
          <span
            class="table__user-status-color table__user-status-color--new"
          />
          <span
            class="table__user-status-value"
          >
            Not Answered
          </span>
        </p>
        <p
          class="table__user-status d-none"
        >
          <span
            class="table__user-status-color table__user-status-color--active"
          />
          <span
            class="table__user-status-value"
          >
            Correct
          </span>
        </p>
        <p
          class="table__user-status d-none"
        >
          <span
            class="table__user-status-color table__user-status-color--blocked"
          />
          <span
            class="table__user-status-value"
          >
            Wrong
          </span>
        </p>
      </td>
      <td>
        <button
          aria-controls="example-collapse-text"
          aria-expanded="false"
        >
          <svg
            class="table__user-down-arrow"
          >
            downArrowResults.svg
          </svg>
        </button>
      </td>
    </tr>
    <tr>
      <td
        class="p-0"
        colspan="5"
      >
        <div
          class="collapse"
        >
          <div
            class="table__row-collapsed"
          >
            <div
              class="test-result__table-collapse-info test-result__table-collapse-info--not-correct"
            >
              <h2
                class="test__info-heading"
              >
                Questions 1-10 are based on the following passage.
              </h2>
              <p>
                Akira came directly, breaking all tradition. Was that it? Had he followed form had he asked his mother to speak to his father to approach a Line go-between would Chie have been more receptive? He came on a winter’s eve. He pounded on the door while a cold rain beat on the shuttered veranda, so at first Chie thought him only the wind. The maid knew better. Chie heard her soft scuttling footsteps, the creak of the door. Then the maid brought a 10 calling card to the drawing room, for Chie.
              </p>
              <p>
                Chie was reluctant to go to her guest; perhaps she was feeling too cozy. She and Naomi were reading at a low table set atop a charcoal brazier. A thick quilt spread over the sides of the table so their legs were 15 tucked inside with the heat. “Who is it at this hour, in this weather?” Chie questioned as she picked the name card off the maid’s lacquer tray.
              </p>
              <p>
                Chie was reluctant to go to her guest; perhaps she was feeling too cozy. She and Naomi were reading at a low table set atop a charcoal brazier. A thick quilt spread over the sides of the table so their legs were 15 tucked inside with the heat. “Who is it at this hour, in this weather?” Chie questioned as she picked the name card off the maid’s lacquer tray.
              </p>
              <p>
                Chie was reluctant to go to her guest; perhaps she was feeling too cozy. She and Naomi were reading at a low table set atop a charcoal brazier. A thick quilt spread over the sides of the table so their legs were 15 tucked inside with the heat. “Who is it at this hour, in this weather?” Chie questioned as she picked the name card off the maid’s lacquer tray.
              </p>
            </div>
            <div
              class="test-result__table-collapse-question test-result__table-collapse-question--not-correct"
            >
              <h2
                class="test__question-heading"
              >
                Which choice best describes what happens in the passage?
              </h2>
              <div
                class="test__input-answer"
              />
              <div
                class="test__choices-wrapper"
              >
                <div
                  class="test__choice-wrapper test__choice-wrapper--correct"
                >
                  <span
                    class="test__choice-alpha"
                  >
                    A- 
                  </span>
                  One Character Argues With Another Character Who Intrudes On
                </div>
                <div
                  class="test__choice-wrapper"
                >
                  <span
                    class="test__choice-alpha"
                  >
                    B- 
                  </span>
                  One Character Argues With Another Character Who Intrudes On
                </div>
                <div
                  class="test__choice-wrapper test__choice-wrapper--not-correct"
                >
                  <span
                    class="test__choice-alpha"
                  >
                    C- 
                  </span>
                  One Character Argues With Another Character Who Intrudes On
                </div>
                <div
                  class="test__choice-wrapper"
                >
                  <span
                    class="test__choice-alpha"
                  >
                    D- 
                  </span>
                  One Character Argues With Another Character Who Intrudes On
                </div>
              </div>
            </div>
          </div>
        </div>
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
